helm_deploy/django-app/templates/deployment.yaml

```diff
@@ -29,22 +29,13 @@ spec:
               fieldRef:
                 fieldPath: status.podIP
           - name: SECRET_KEY
-            valueFrom:
-              secretKeyRef:
-                name: {{ template "django-app.fullname" . }}
-                key: secretKey
+            value: {{ .Values.deploy.secretKey | quote }}
```

Comment:
This might be me not explaining myself very well, but we still want to use a `Secret` object for all of these values and a `secretKeyRef` here, but just with the value interpolated from the `values.yml` files, e.g:

```
apiVersion: v1
kind: Secret
metadata:
  name: {{ template "django-app.fullname" . }}
type: Opaque
data:
  secretKey: {{ .Values.deploy.secretKey | b64enc | quote }}
```

Note the use of the `b64enc` helper that to take care of base64 encoding.